dice_ml/counterfactual_explanations.py

```diff
@@ -127,11 +92,134 @@ def visualize_as_list(self, display_sparse_df=True,
     def to_json(self):
         """ Serialize Explanations object to json.
         """
-        return json.dumps(self, default=json_converter,
-                          indent=2)
+        serialization_version = _CurrentVersion
+        if serialization_version == '1.0':
+            cf_examples_str_list = []
+            for cf_examples in self.cf_examples_list:
+                cf_examples_str = cf_examples.to_json(
+                    serialization_version=serialization_version)
+                cf_examples_str_list.append(cf_examples_str)
+            entire_dict = {
+                'cf_examples_list': cf_examples_str_list,
+                'local_importance': self.local_importance,
+                'summary_importance': self.summary_importance,
+                'metadata': self.metadata
+            }
+            return json.dumps(entire_dict)
+        elif serialization_version == '2.0':
+            combined_test_instance_list = []
+            combined_final_cfs_list = []
+            data_interface = None
+            feature_names = None
+            feature_names_including_target = None
+            model_type = None
+            desired_class = None
+            desired_range = None
+            for cf_examples in self.cf_examples_list:
+                cf_examples_str = cf_examples.to_json(
```

Comment:
why do we convert to json and reload from str, `to_json` and then `loads`?

Reply:
This is because the diverse_counterfactusls.py returns a json string and we need to reload it to decompose the json dict according to the new schema